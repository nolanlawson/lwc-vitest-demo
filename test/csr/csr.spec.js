import { createElement } from 'lwc'
import { describe, it, expect } from 'vitest'
import XButton from 'x/button'

describe('basic test suite', () => {
  it('renders stylesheets', async () => {
    const button = createElement('x-button', { is: XButton })
    document.body.appendChild(button)
    await Promise.resolve()
    const adoptedOrInlineStylesheets = button.adoptedStyleSheets ?? button.shadowRoot.querySelectorAll('style')
    expect(adoptedOrInlineStylesheets.length).toBe(1, 'Should render a stylesheet')
  })
})
