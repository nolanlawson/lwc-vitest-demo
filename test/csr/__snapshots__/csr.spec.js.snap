// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`basic test suite > clicks the button on the client side 1`] = `
<x-button>
    <template shadowroot="open">
        <span>
            <code>
                Is this SSR?
            </code>
            : false
        </span>
        <span>
            <code>
                Was I clicked?
            </code>
            : true
        </span>
        <button class="pretty">
            I am a button!
        </button>
    </template>
</x-button>
`;

exports[`basic test suite > renders a button on the client side 1`] = `
<x-button>
    <template shadowroot="open">
        <span>
            <code>
                Is this SSR?
            </code>
            : false
        </span>
        <span>
            <code>
                Was I clicked?
            </code>
            : false
        </span>
        <button class="pretty">
            I am a button!
        </button>
    </template>
</x-button>
`;
